pkg/operator/ceph/cluster/mon/health.go

```diff
@@ -237,10 +237,26 @@ func (c *Cluster) failMon(monCount, desiredMonCount int, name string) {
 			logger.Errorf("failed to remove mon %q. %v", name, err)
 		}
 	} else {
+		if c.spec.DisruptionManagement.ManagePodBudgets {
+			logger.Info("prevent voluntary mon drain while failing over")
+			err := blockMonDrain(c.context, c.Namespace)
+			if err != nil {
+				logger.Errorf("failed to block mon drain. %v", err)
+			}
+		}
+
 		// bring up a new mon to replace the unhealthy mon
 		if err := c.failoverMon(name); err != nil {
 			logger.Errorf("failed to failover mon %q. %v", name, err)
 		}
+
+		if c.spec.DisruptionManagement.ManagePodBudgets {
```

Comment:
Just because we only log errors, I believe the commit should explain that we are in best-effort mode.

Reply:
updated commit message to indicate that we are using best effort.